### PR TITLE
CO-3254 Implement resubmission of unsuccesful translation check

### PR DESCRIPTION
**Pull request by SylvainLosey**
I had a hard time testing this, not entirely sure about

```
# Concatenate the local_id and the current resubmit_id
if "local_id" in odoo_data:
    odoo_data["local_id"] = odoo_data["local_id"] + self.resubmit_id
```

as I didn't find a way to trigger a letter json_to_data from GMC

### CODE REVIEW

File: sbc_compassion/models/correspondence.py [hunk 1]
```diff
@@ -965,4 +966,18 @@ def json_to_data(self, json, mapping_name=None):
                 pages.append(orm_tuple)
             odoo_data["page_ids"] = pages or False
 
+        # Concatenate the local_id and the current resubmit_id
+        if "local_id" in odoo_data:
+            odoo_data["local_id"] = odoo_data["local_id"] + self.resubmit_id
+
```
Comment (ecino) [inline sbc_compassion/models/correspondence.py]: No, this is in the wrong method, you should do this in the method `data_to_json` to change the value we are sending to GMC.

File: sbc_compassion/models/correspondence.py [hunk 2]
```diff
@@ -965,4 +966,18 @@ def json_to_data(self, json, mapping_name=None):
...
+    @api.multi
+    def resubmit_letter_translation(self):
+        for letter in self:
+            if letter.state != "Translation check unsuccessful":
+                raise UserError(_("Letter must be in state 'Translation check unsuccessful'"))
+
+            letter.kit_identifier = None
+            letter.send_local_translate()
+            letter.resubmit_id += 1
```
Comment (ecino) [inline sbc_compassion/models/correspondence.py]:
```suggestion
            letter.resubmit_id += 1
            letter.create_commkit()
```

File: sbc_compassion/views/correspondence_view.xml [hunk 3]
```diff
@@ -32,6 +32,14 @@
         </field>
     </record>
 
+    <!-- Resubmit letter translation mass action-->
+    <record id="resubmit_letter_translation_action" model="ir.actions.server">
+        <field name="name">Resubmit letter translation</field>
```
Comment (ecino) [inline sbc_compassion/views/correspondence_view.xml]:
```suggestion
        <field name="name">Resubmit letter to GMC</field>
```

File: sbc_compassion/views/correspondence_view.xml [hunk 4]
```diff
@@ -32,6 +32,14 @@
         </field>
     </record>
 
+    <!-- Resubmit letter translation mass action-->
+    <record id="resubmit_letter_translation_action" model="ir.actions.server">
```
Comment (ecino) [inline sbc_compassion/views/correspondence_view.xml]:
```suggestion
    <record id="resubmit_letter_action" model="ir.actions.server">
```

File: sbc_compassion/views/correspondence_view.xml (hunk 3 above)
Comment (ecino) [inline sbc_compassion/views/correspondence_view.xml]:
```suggestion
        <field name="name">Resubmit letter</field>
```

File: sbc_compassion/views/correspondence_view.xml [hunk 5]
```diff
@@ -32,6 +32,14 @@
...
+        <field name="name">Resubmit letter translation</field>
+        <field name="model_id" ref="sbc_compassion.model_correspondence"/>
+        <field name="state">code</field>
+        <field name="code">records.resubmit_letter_translation()</field>
```
Comment (ecino) [inline sbc_compassion/views/correspondence_view.xml]:
```suggestion
        <field name="code">records.resubmit_letter()</field>
```

File: sbc_compassion/views/correspondence_view.xml [hunk 6]
```diff
@@ -52,6 +60,9 @@
                     <button name="download_pdf" type="object"
                             string="Download PDF"
                             attrs="{'invisible': [('direction', '!=', 'Supporter To Beneficiary')]}"/>
+                    <button name="resubmit_letter_translation" type="object"
```
Comment (ecino) [inline sbc_compassion/views/correspondence_view.xml]:
```suggestion
                    <button name="resubmit_letter" type="object"
```

File: sbc_compassion/views/correspondence_view.xml [hunk 7]
```diff
@@ -52,6 +60,9 @@
...
+                            string="Resubmit Letter Translation"
```
Comment (ecino) [inline sbc_compassion/views/correspondence_view.xml]:
```suggestion
                            string="Resubmit Letter to GMC"
```

File: sbc_compassion/views/correspondence_view.xml (hunk 4 above)
Comment (ecino) [inline sbc_compassion/views/correspondence_view.xml]:
```suggestion
    <record id="resubmit_letter_action" model="ir.actions.server">
```

File: sbc_compassion/models/correspondence.py [hunk 8]
```diff
@@ -965,4 +966,18 @@ def json_to_data(self, json, mapping_name=None):
...
             odoo_data["page_ids"] = pages or False
 
+        # Concatenate the local_id and the current resubmit_id
+        if "local_id" in odoo_data:
+            odoo_data["local_id"] = odoo_data["local_id"] + self.resubmit_id
+
         return odoo_data
+
+    @api.multi
+    def resubmit_letter_translation(self):
```
Comment (ecino) [inline sbc_compassion/models/correspondence.py]:
```suggestion
    def resubmit_letter(self):
```

**Review by ecino — CHANGES_REQUESTED** [review]

**Comment by SylvainLosey** [comment]
Requested changes applied

**Review by ecino — APPROVED** [review]

**Pull request merged**